src/main/java/seedu/address/model/good/Good.java

```diff
@@ -21,16 +23,27 @@
     private final GoodName goodName;
     private final GoodQuantity goodQuantity;
     private final GoodQuantity threshold;
+    private final Name supplierName;
 
-    public Good(GoodName goodName, GoodQuantity goodQuantity, GoodQuantity threshold) {
+
+    public Good(GoodName goodName, GoodQuantity goodQuantity, GoodQuantity threshold, Name supplierName) {
```

Comment:
Should Good contains supplierName? if there are multiple supplier selling the same Good, then which supplier name should the good contain?

Reply:
This would be the good created to encapsulate information on 1 specific BuyCommand. The supplier name is not persistent once the BuyCommand has been processed. I agree that this is might not be very clear, I will try to think of an alternate approach, possibly using your "buy via displayed index" method.

Reply:
if you need the supplier name in the buy command, you may want to change the buy command constructor from
(Good boughtGood) to (Good boughtGood, Name supplierName).